client/src/hooks/addEvent.ts

```diff
@@ -0,0 +1,17 @@
+export default async function addEvent(data: FormData) {
+  try {
+    const response = await fetch("http://localhost:8000/api/event/", {
```

Comment:
Instead of the base fetch, use the axios client which you can import as `import api from "@/lib/api";`. This will allow you do authenticate the requests easily